lib/postgrex/types.ex

```diff
@@ -108,8 +108,8 @@ defmodule Postgrex.Types do
       :ets.insert(table, {extension, opts})
     end
 
-    for type_info <- types,
-        extension = find_extension(type_info, extensions, extension_opts, oid_types) do
+    for type_info <- types do
+      extension = find_extension(type_info, extensions, extension_opts, oid_types)
```

Comment:
`find_extension/4` is handling the case where `type_info` is nil. We need to handle that here too or `type_info.oid` will crash. If it can never be nil, then remove the clause from  `find_extension/4`.